modules/instances/main.tf

```
@@ -0,0 +1,134 @@
+# Security group for public instance
+resource "aws_security_group" "public_instance_sg" {
+  name        = "tf-public-instance-sg"
+  description = "Security group for public instance"
+  vpc_id      = var.vpc_id
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = var.public_instance_sg_ssh_inbound
+  }
+
+  egress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = var.public_instance_sg_https_outbound
+  }
+
+  egress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = var.public_instance_sg_ssh_outbound
+  }
+}
+
+#Security group for private instance
+resource "aws_security_group" "private-instance-sg" {
```

Have a look at the article about naming conventions linked in another comment
```suggestion
resource "aws_security_group" "private_instance_sg" {
```